app/src/main/java/org/mozilla/focus/screenshot/ScreenshotManager.java

```diff
@@ -52,4 +69,47 @@ public void update(org.mozilla.focus.screenshot.model.Screenshot screenshot, Asy
     public void query(int offset, int limit, AsyncQueryListener listener) {
         mQueryHandler.startQuery(QueryHandler.SCREENSHOT_TOKEN, listener, Uri.parse(Screenshot.CONTENT_URI.toString() + "?offset=" + offset + "&limit=" + limit), null, null, null, Screenshot.TIMESTAMP + " DESC");
     }
+
+    private void initScreenShotCateogry(Context context) {
+        if (categories.size() != 0) {
+            return;
+        }
+        final JSONObject json;
+        try {
+            json = IOUtils.readAsset(context, "screenshots-mapping.json");
+
+            final Iterator<String> iterator = json.keys();
+            while (iterator.hasNext()) {
+                final String category = iterator.next();
+                final Object o = json.get(category);
+                if (o instanceof JSONArray) {
+                    JSONArray array = ((JSONArray) o);
+                    for (int i = 0; i < array.length(); i++) {
+                        final Object domain = array.get(i);
+                        if (domain instanceof String) {
+                            categories.put((String) domain, category);
+                        }
+                    }
+                }
+            }
+        } catch (IOException | JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public String getCategory(String url) {
+        try {
+            if (categories.size() == 0) {
+                return "";
+            }
```

Comment:
there might be threading issue here. And here there are two cases: if no categories we return `""` empty string, if any exception we return `DEFAULT_CATEGORY`. Leave a comment for the difference would be helpful.